### ISSUE TITLE
Can we increase the speed of the model by reducing the number of default boxes?

### ISSUE THREAD

**Issue by Goldit** [weiliu89/caffe#290]
Hi all,

I removed the conv4_3 and fc7 layers from the mbox_source_layers and trained a new model. However, the speed of the new model and the original one were quite similar?

Because the conv4_3 and fc7 layers generate around 90% of the default boxes (18051 / 20097 default boxes), should we expect an increase in speed by removing the default boxes from the two layers?

Please see the attached files for more details on the benchmarks of the two models.

[new_model_benchmark.txt](https://github.com/weiliu89/caffe/files/594352/new_model_benchmark.txt)
[original_model_benchmark.txt](https://github.com/weiliu89/caffe/files/594353/original_model_benchmark.txt)

Best Regards,
Nhan.

**Comment by weiliu89** [comment]
@Goldit I don't think that can increase speed much. If you use VGGNet as base network, more than 80% of the computation is spent on VGG. And the time for all extra layers and all prediction layers is same as the nms step. You can use `./build/tools/caffe time` to check the speed for each layer.

If you want to increase speed, you could try to use a different base network.
